Content.Server/GameObjects/Components/Weapon/Melee/MeleeChemicalInjectorComponent.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Content.Server.GameObjects.Components.Body.Circulatory;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Shared.Chemistry;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.Weapon.Melee
+{
+    [RegisterComponent]
+    public class MeleeChemicalInjectorComponent : Component
+    {
+        public override string Name => "MeleeChemicalInjector";
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public ReagentUnit TransferAmount { get; set; }
+
+        public float TransferEfficiency { get => _transferEfficiency; set => _transferEfficiency = Math.Clamp(value, 0, 1); }
+        private float _transferEfficiency;
+
+        [ViewVariables]
+        private SolutionContainerComponent _solutionContainer;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(this, x => TransferAmount, "transferAmount", ReagentUnit.New(1));
+            serializer.DataField(ref _transferEfficiency, "transferEfficiency", 1f);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            var meleeWeapon = Owner.EnsureComponent<MeleeWeaponComponent>();
+            meleeWeapon.OnHittingEntities += OnMeleeHit;
+            _solutionContainer = Owner.EnsureComponent<SolutionContainerComponent>();
+        }
+
+        private void OnMeleeHit(IEnumerable<IEntity> hitEntities)
+        {
+            foreach (var entity in hitEntities)
+            {
+                if (!entity.TryGetComponent<BloodstreamComponent>(out var bloodstream))
+                    return;
+                var removedSolution = _solutionContainer.Solution.SplitSolution(TransferAmount);
+                var removedVolume = removedSolution.TotalVolume;
+                var solutionToInject = removedSolution.SplitSolution(removedVolume * TransferEfficiency);
+                bloodstream.TryTransferSolution(solutionToInject);
```

Comment:
Should this split the transferamount equally by the hit entities? You could also check if there's any solution remaining up front to avoid doing the rest (e.g. if it hits 3 entities and depletes after one).

Reply:
I've made it split the transferred solution equally between targets to better handle the case when there is not enough for multiple hit targets.